app/models/application_record.rb

```diff
@@ -62,6 +62,12 @@ def self.with_statement_timeout(duration, connection: self.connection)
     connection.execute "SET statement_timeout = #{milliseconds}"
   end
 
+  def self.unique_across_models(attribute, **options)
+    CrossModelSlug.register(self, attribute)
+    validates attribute, presence: true
+    validates attribute, cross_model_slug: true, **options, if: :"#{attribute}_changed?"
+  end
```

Comment:
nit/question: Would it make more sense for this to live as a concern instead? I suppose that's just a pattern I'm more familiar with.

Reply:
I think either works, and `concerns` would probably be a better fit. I'll move it there. (I am old, I forget `concerns` exists. Rails ... 4 (!) is still very "new" to my old brain. :brain: )

Reply:
I think this was fine. As far as i know, code should be moved to concern only if it will be used in more than one class/module.

Reply:
Sorry I should've clarified the concern I'm thinking about is something like this

```ruby
Class User
  includes Slugable
end

Class Article
  includes Slugable
end

module Slugable
  extend ActiveSupport::Concern

  included do
    validates attribute, presence: true
    validates attribute, cross_model_slug: true, **options, if: :"#{attribute}_changed?"
  end
end
```

Reply:
@maestromac and I wound up chatting about this a bit yesterday. The proposed pseudo-code above doesn't work in this case, because not all of these models use `slug` as the attribute name (`User#username` is the "problem"), that's what made me think a macro would work here. 

I'm pretty open to suggestions on exactly where this lives, but I have a pedantic desire to avoid `include`ing a module who's only job is adding a class-level macro.